src/shared/sleep-config.c

```diff
@@ -368,8 +395,8 @@ int find_hibernate_location(HibernateLocation **ret_hibernate_location) {
                     || ((swap->priority == hibernate_location->swap->priority)
                         && (swap->size - swap->used) > (hibernate_location->swap->size - hibernate_location->swap->used))) {
 
-                        _cleanup_free_ char *swap_device_id = NULL;
-                        r = swap_device_to_major_minor(swap, &swap_device_id);
+                        dev_t swap_device;
+                        r = swap_device_to_device_id(swap, &swap_device);
                         if (r < 0)
```

Comment:
something should log if this call fails

Reply:
Updated. Now swap_device_to_device_id() doesn't log at all, and the caller logs.